azurerm/resource_arm_dns_srv_record.go

```diff
@@ -77,154 +77,138 @@ func resourceArmDnsSrvRecord() *schema.Resource {
 	}
 }
 
-func resourceArmDnsSrvRecordCreate(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
+func resourceArmDnsSrvRecordCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	dnsClient := meta.(*ArmClient).dnsClient
 
-	tags := d.Get("tags").(map[string]interface{})
-	expandedTags := expandTags(tags)
-
-	createCommand := &dns.CreateSRVRecordSet{
-		Name:              d.Get("name").(string),
-		Location:          "global",
-		ResourceGroupName: d.Get("resource_group_name").(string),
-		ZoneName:          d.Get("zone_name").(string),
-		TTL:               d.Get("ttl").(int),
-		Tags:              *expandedTags,
-	}
-
-	srvRecords, recordErr := expandAzureRmDnsSrvRecord(d)
-	if recordErr != nil {
-		return fmt.Errorf("Error Building Azure RM SRV Record: %s", recordErr)
-	}
-	createCommand.SRVRecords = srvRecords
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	zoneName := d.Get("zone_name").(string)
+	ttl := int64(d.Get("ttl").(int))
 
-	createRequest := rivieraClient.NewRequest()
-	createRequest.Command = createCommand
+	tags := d.Get("tags").(map[string]interface{})
+	metadata := expandTags(tags)
 
-	createResponse, err := createRequest.Execute()
-	if err != nil {
-		return fmt.Errorf("Error creating DNS SRV Record: %s", err)
-	}
-	if !createResponse.IsSuccessful() {
-		return fmt.Errorf("Error creating DNS SRV Record: %s", createResponse.Error)
+	records, err := expandAzureRmDnsSrvRecords(d)
+	props := dns.RecordSetProperties{
+		Metadata:   metadata,
+		TTL:        &ttl,
+		SrvRecords: &records,
 	}
 
-	readRequest := rivieraClient.NewRequest()
-	readRequest.Command = &dns.GetSRVRecordSet{
-		Name:              d.Get("name").(string),
-		ResourceGroupName: d.Get("resource_group_name").(string),
-		ZoneName:          d.Get("zone_name").(string),
+	parameters := dns.RecordSet{
+		Name:                &name,
+		RecordSetProperties: &props,
 	}
 
-	readResponse, err := readRequest.Execute()
+	//last parameter is set to empty to allow updates to records after creation
+	// (per SDK, set it to '*' to prevent updates, all other values are ignored)
+	resp, err := dnsClient.CreateOrUpdate(resGroup, zoneName, name, dns.SRV, parameters, "", "")
```

Comment:
what's the penultimate option for here? generally when there's more than one empty argument we'd tend to separate these out into variables above, i.e.:
```
foo := "bar"
bar := "foo"
...CreateOrUpdate(...., foo, bar)
```